matchbox/resource_profile.go

```diff
@@ -80,14 +86,28 @@ func resourceProfileCreate(d *schema.ResourceData, meta interface{}) error {
 	for _, arg := range d.Get("args").([]interface{}) {
 		args = append(args, arg.(string))
 	}
-	profile := &storagepb.Profile{
-		Id:         name,
-		IgnitionId: clcName,
-		Boot: &storagepb.NetBoot{
-			Kernel: d.Get("kernel").(string),
-			Initrd: initrds,
-			Args:   args,
-		},
+
+	profile := &storagepb.Profile{}
+	if strings.HasSuffix(tmplName, ".generic") {
```

Comment:
Let's gate based on wither generic contents were provided, no based on a suffix side effect of how the naming is being done below